src/main/java/build/buildfarm/instance/AbstractServerInstance.java

```diff
@@ -322,17 +325,19 @@ protected void validateAction(Action action) {
     // invalid action?
     filesUniqueAndSortedPrecondition(action.getOutputFilesList());
     filesUniqueAndSortedPrecondition(action.getOutputDirectoriesList());
-    Command command;
-    try {
-      command = Command.parseFrom(getBlob(commandDigest));
-    } catch (InvalidProtocolBufferException ex) {
-      Preconditions.checkState(
-          false,
-          INVALID_DIGEST);
-      return;
+    if (commandDigest.getSizeBytes() != 0 || Command.getDefaultInstance().toByteString().size() != 0) {
```

Comment:
I think an empty command is a good reason for an INVALID_ARGUMENT.

Reply:
I'll buy that as well, since we don't believe any representation of a message and its [lack of] contents are valid arbitrarily.